### PR TITLE
Feat/sample app

### PR DESCRIPTION
**Pull request by gorbuvla**
Sample app to showcase usage, copied from RxOAuth.

RxJava is removed, but no couroutine calls are implemented yet since it is a subject of next PR.

### CODE REVIEW

File: app/src/main/java/cz/ackee/sample/di/DIContainer.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+
+/**
+ * Simple DI container that provides dependencies
+ */
+class DIContainer(val app: App) {
+
+//    val rxOAuthManager = RxOAuthManager(
```
Comment (davidbilik) [inline app/src/main/java/cz/ackee/sample/di/DIContainer.kt:21]: can be removed?

File: app/src/main/java/cz/ackee/sample/di/DIContainer.kt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            .addConverterFactory(MoshiConverterFactory.create())
+
+    val authApiDescription = retrofitBuilder
+        .addCallAdapterFactory(AckroutineCallAdapterFactory())
+        .build()
+        .create(AuthApiDescription::class.java)
+
+    val apiDescription: ApiDescription = retrofitBuilder
+        .client(OkHttpClient.Builder()
+            //.addNetworkInterceptor(rxOAuthManager.provideAuthInterceptor())
```
Comment (davidbilik) [inline app/src/main/java/cz/ackee/sample/di/DIContainer.kt:47]: same?

File: app/src/main/java/cz/ackee/sample/interactor/IApiInteractor.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import cz.ackee.ackroutine.core.OAuthCredentials
+import cz.ackee.sample.model.SampleItem
+import kotlinx.coroutines.Deferred
+
+/**
+ * Interactor for communicating with API
+ */
+interface IApiInteractor {
+
+    fun getData(): Deferred<List<SampleItem>>
```
Comment (davidbilik) [inline app/src/main/java/cz/ackee/sample/interactor/IApiInteractor.kt:12]: In our applications we are trying to have Deferred only on ApiDescription level and Interactor has "hard data" as in
`suspend fun getData():List<SampleItem>`. Does it make sense here to make it the same?

File: app/src/main/java/cz/ackee/sample/di/DIContainer.kt (hunk 1 above)
Comment (gorbuvla) [reply app/src/main/java/cz/ackee/sample/di/DIContainer.kt:21]: next PR focuses on coroutine integration of all the commented out stuff, this PR is just a hard copy of RxOAuth example without rxjava/rxoauth dependency

File: app/src/main/java/cz/ackee/sample/di/DIContainer.kt (hunk 2 above)
Comment (gorbuvla) [reply app/src/main/java/cz/ackee/sample/di/DIContainer.kt:47]: yep, same :)

File: app/src/main/java/cz/ackee/sample/interactor/IApiInteractor.kt (hunk 3 above)
Comment (gorbuvla) [reply app/src/main/java/cz/ackee/sample/interactor/IApiInteractor.kt:12]: i will take a look at it and push changes to more recent branch with workin example app

**Pull request merged**